flight/Modules/GPS/GPS.c

```diff
@@ -261,20 +260,22 @@ static void gpsTask(void *parameters)
 			}
 
 			if (res == PARSER_COMPLETE) {
-				timeNowMs = PIOS_Thread_Systime();
-				timeOfLastUpdateMs = timeNowMs;
-				timeOfLastCommandMs = timeNowMs;
+				timeOfLastUpdateMs = PIOS_Thread_Systime();
 			}
 		}
 
 		// Check for GPS timeout
-		timeNowMs = PIOS_Thread_Systime();
+		uint32_t timeNowMs = PIOS_Thread_Systime();
 		if ((timeNowMs - timeOfLastUpdateMs) >= GPS_TIMEOUT_MS) {
 			// we have not received any valid GPS sentences for a while.
 			// either the GPS is not plugged in or a hardware problem or the GPS has locked up.
 			uint8_t status = GPSPOSITION_STATUS_NOGPS;
 			GPSPositionStatusSet(&status);
 			AlarmsSet(SYSTEMALARMS_ALARM_GPS, SYSTEMALARMS_ALARM_ERROR);
+			/* Don't reinitialize too often. */
+			if ((timeNowMs - timeOfLastCommandMs) >= GPS_TIMEOUT_MS) {
```

Comment:
So, I slightly increased the timeout from the default 500ms in the original pull. It requires not receiving any GPS messages for 750ms and being 750ms from the last init attempt.

If this ever happens in flight, 750ms is a long, long time.  We expect to receive a new set of nav messages every 100ms or 200ms (and several messages during each update).   A delay of 600ms+ should never happen without a failure, and I don't think reiniting takes too much resources / makes anything worse.